linera-base/src/task.rs

```diff
@@ -55,12 +55,15 @@ mod implementation {
         recv
     }
 
-    /// Spawns a blocking task on the same thread (!).
-    /// TODO(#2399): replace this by a Web worker.
+    /// Spawns a blocking task on a new Web Worker.
     pub fn spawn_blocking<R: Send + 'static, F: FnOnce() -> R + Send + 'static>(
         task: F,
     ) -> BlockingFuture<R> {
-        spawn(async { task() })
+        let (send, recv) = oneshot::channel();
+        wasm_thread::spawn(move || {
+            let _ = send.send(task());
```

Comment:
What is the `let _ =` for? Are you suppressing a warning intentionally? (If yes, add a comment. Otherwise, I'd suggest not to use `let _ =` in the first place.)